tests/test_model_training.py

```diff
@@ -38,6 +38,31 @@ def test_model_training(shapes_image_size, shapes_sample, shapes_sample_config,
     assert score.shape == (BATCH_SIZE, len(label_ids))
     assert all(0 <= s and s <= 1 for s in score.ravel())
 
-    BACKUP_FILENAME = os.path.join(str(shapes_temp_dir), "checkpoints", "test_model.h5")
+    BACKUP_FILENAME = os.path.join(str(shapes_temp_dir), "checkpoints", "test_model_{:02d}.h5".format(NB_EPOCHS))
     model.save(BACKUP_FILENAME)
     assert os.path.isfile(BACKUP_FILENAME)
+
+def test_model_backup_loading(shapes_image_size, shapes_sample_config, shapes_temp_dir):
+    """Test the model checkpoint recovering
+    """
+    config = read_config(shapes_sample_config)
+    label_ids = [x['id'] for x in config['labels'] if x['is_evaluate']]
+
+    cnn = FeatureDetectionNetwork("test", image_size=shapes_image_size, nb_labels=len(label_ids))
+    model = Model(cnn.X, cnn.Y)
+    old_weights = model.get_weights()
+    checkpoint_path = os.path.join(str(shapes_temp_dir), "checkpoints")
+    if os.path.isdir(checkpoint_path):
+        checkpoints = os.listdir(checkpoint_path)
+        if len(checkpoints) > 0:
+            model_checkpoint = checkpoints[-1]
+            trained_model_epoch = int(model_checkpoint[-5:-3])
+            checkpoint_complete_path = os.path.join(checkpoint_path, model_checkpoint)
+            model.load_weights(checkpoint_complete_path)
+    else:
+        trained_model_epoch = 0
+    new_weights = model.get_weights()
+    assert trained_model_epoch > 0
+    assert len(old_weights) == len(new_weights)
+    assert old_weights[0].shape == new_weights[0].shape
+    assert all(not np.allclose(old_weights[i], new_weights[i]) for i in range(len(new_weights)))
```

Comment:
I wonder if we can use a numpy testing assert here, such as `np.testing.assert_almost_equal`

Reply:
Why not, I do not know this possibility. I was wondering if the most accurate is really `assert_almost_equal`, as we handle integers. I see a `assert_allclose` which is close from the current used function `np.allclose`.

Reply:
Yark, we do not test if all weights are equivalent. The goal is the opposite one: to check if each new weight is different from each old weight. It seems that there is no convenient numpy test for such a task.

I tried a workaround with a `pytest` context manager, something like:
```
with pytest.raises(AssertionError):
    any(np.testing.assert_allclose(old_weights[i], new_weights[i]) for i in range(len(new_weights)))
```
However it is not clear for me that this snippet is really accurate (*e.g.* the test result is the same if I write `any(not ...)`).